fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/lifecycle/v1/ClientLifecycleEvents.java

```diff
@@ -51,13 +52,39 @@ private ClientLifecycleEvents() {
 		}
 	});
 
+	/**
+	 * Called when the current player in the Minecraft game is moved to a different world.
+	 * This occurs before the client's {@link MinecraftClient#world current world} is disposed and a new world is created.
+	 */
+	public static final Event<ChangeWorld> CHANGE_WORLD = EventFactory.createArrayBacked(ChangeWorld.class, callbacks -> (client, world, destination) -> {
+		for (ChangeWorld callback : callbacks) {
+			callback.onChangeWorld(client, world, destination);
+		}
+	});
+
+	@Environment(EnvType.CLIENT)
 	@FunctionalInterface
 	public interface ClientStarted {
 		void onClientStarted(MinecraftClient client);
 	}
 
+	@Environment(EnvType.CLIENT)
 	@FunctionalInterface
 	public interface ClientStopping {
 		void onClientStopping(MinecraftClient client);
 	}
+
+	@Environment(EnvType.CLIENT)
+	@FunctionalInterface
+	public interface ChangeWorld {
+		/**
+		 * Called when the current player in the Minecraft game is moved to a different world.
```

Comment:
```suggestion
		 * Called when the current player in the Minecraft game has moved to a different world.
```
indicate the move is already done